plugins/net_plugin/net_plugin.cpp

```diff
@@ -3598,9 +3597,12 @@ namespace eosio {
             sync_master->sync_recv_block( c, blk_id, blk_num, true );
          });
       } else {
-         c->strand.post( [sync_master = my_impl->sync_master.get(), dispatcher = my_impl->dispatcher.get(), c, blk_id, blk_num]() {
-            sync_master->rejected_block( c, blk_num );
-            dispatcher->rejected_block( blk_id );
+         c->strand.post( [c, blk_id, blk_num, reason]() {
+            if( reason == unlinkable ) {
+               my_impl->dispatcher->rm_peer_block( blk_id, c->connection_id );
```

Comment:
what about just move `my_impl->dispatcher->add_peer_block( blk_id, c->connection_id );` beyond `if ( !accepted )` ?
I don't see any impact of this and you won't need to remove it in case of failure

Reply:
Could, but I first missed the need for the call. I prefer to have it called from one spot instead of "hidden" in the try-catch.

Reply:
make sense then. Missed that we have few catch blocks.